src/dymaptic.GeoBlazor.Core/Components/Layers/ImageryLayer.cs

```diff
@@ -0,0 +1,148 @@
+﻿using dymaptic.GeoBlazor.Core.Components.Renderers;
+using dymaptic.GeoBlazor.Core.Exceptions;
+using dymaptic.GeoBlazor.Core.Serialization;
+using Microsoft.AspNetCore.Components;
+using System.Text.Json.Serialization;
+
+
+namespace dymaptic.GeoBlazor.Core.Components.Layers;
+
+/// <summary>
+///     Represents a dynamic image service resource as a layer. An ImageryLayer retrieves and displays data from dynamic image services. An image service supports
+///     accessing the mosaicked image, its catalog, and the individual rasters in the catalog. An image service supports dynamic access and tiled access. 
+///     <a target="_blank" href="https://developers.arcgis.com/javascript/latest/api-reference/esri-layers-ImageryLayer.html">
+///         ArcGIS
+///         JS API
+///     </a>
+/// </summary>
+public class ImageryLayer : Layer
+{
+    /// <inheritdoc />
+    [JsonPropertyName("type")]
+    public override string LayerType => "imagery";
+
+    /// <summary>
+    ///     Parameterless constructor for use as a razor component
+    /// </summary>
+    public ImageryLayer()
+    {
+    }
+
+    /// <summary>
+    ///     Constructor for use in code
+    /// </summary>
+    /// <param name="url">
+    ///     The absolute URL of the REST endpoint of the layer, non-spatial table or service
+    /// </param>  
+    /// <param name="portalItem">
+    ///     The <see cref="PortalItem" /> from which the layer is loaded.
+    /// </param>
+    /// <param name="renderer">
+    ///     An abstract base class that implements a variety of imagery renderers.
+    /// </param>
+
+    public ImageryLayer(string? url = null, PortalItem? portalItem = null, ImageryRenderer? renderer = null
+        ) 
+    {
+        if (url is null && portalItem is null)
```

Comment:
We need the `#pragma warning disable BL0005`, such as you can see on `Point.cs` to silence setting Blazor `[Parameters]` in code in these constructors. It looks like it is missing from all the Layer types, can you add that?

Reply:
Added the Prgma warning to: `ImageryLayer.cs, OpenStreetMapLayer.cs, KMLLayer.cs, GraphicsLayer.cs, GeoRSSLayer.cs, GeoJSONLayer.cs and FeatureLayer.cs`